client/BlockList.jsx

```diff
@@ -14,12 +14,25 @@ class InnerBlockList extends React.Component {
         super(props);
 
         this.state = {
-            username: ''
+            username: '',
+            detailsLoaded: false
         };
     }
 
-    componentDidMount() {
-        this.props.loadBlockList(this.props.user);
+    componentWillMount() {
+        if(this.props.user) {
+            this.props.loadBlockList(this.props.user);
+
+            this.setState({ detailsLoaded: true });
+        }
+    }
+
+    componentWillReceiveProps(props) {
+        if(!this.state.detailsLoaded && props.user) {
+            this.props.loadBlockList(props.user);
```

Comment:
Not quite, because i've done away with the passing of user object/token from server to client on page load (which is a good thing), it does a call to the API on page load.  Will receive props covers that scenario, whereas will mount covers where you are already authed and just navigate to the page